### ISSUE TITLE
customizing bfs_custom.c

### ISSUE THREAD

**Issue by azlefty** [graph500/graph500#9]
I'd like to modify the functions `make_graph_data_structure` and `run_bfs` in **bfs_custom.c** to accommodate my own kernels.  However my c/c++ programming experience is fairly rudimentary and I don't understand how to access the graph data.  Are they in CSR arrays and if so, how can I access them and/or what are the addresses?  I am stepping through the un-optimized code (removing the -03  in eclipse debug mode, but most of the variables seem to be optimized out anyway, so I was not able to get much help from the **bfs_reference.c** versions.

If I could trouble you to provide some sample code that I could insert into those functions that simply access the data and print it to the screen, that would help to get me started.  Thanks very much.

**Comment by akorzh** [comment]
initial data being generated by generator is list of tuples (e1,e2,{w}) weight is optional and needed for SSSP.
Then kernel 1 kicks in and for now it converts it to CSR format.
Then Kernel 2(bfs) and kernel3(sssp) do their jobs having converted CRS graph as input.

If you really want to change kernel1 and use different format for kernels 2 and 3, you should create copy of files csr_reference.{c,h} and modify code inside by creating other structure then oned_csr_graph.

Then you 'll need to edit Makefile a bit to use those new kernel1 files in conjuction with your bfs_custom.c and sssp_custom.c code which would input new structure ( it's transferred as top-level global btw) .  make_graph_data_structure  is just a wrapper which calls actual kernel1 from csr_reference.c

Also don't forget to disable flag -USE_CRS_FOR_VALIDATION if your CRS would be different.

hope this helps.

**Comment by azlefty** [comment]
Thanks, I will try to look at that.

Meanwhile, I was able to  get the stand-alone Kroenecker graph generator in the generator folder, written by Jeremiah Willcock and Andrew Lumsdaine, running.  Is that still being supported?   

Regarding the Willcock / Lumsdaine generator:  I noticed for example that it appears to generate pairs for a directed graph with self-loops.  I had understood the benchmark to be an undirected graph without self-loops.  I can add that but I wanted to confirm that wasn't already an option. 

Also: Have the vertex id's already been scrambled?
